### DISCUSSION TITLE
Skim Matrix Variable Explaination

### DISCUSSION THREAD

**Discussion by HuyLeDinhGia** [synthicity/activitysim#912] [General]
Hi ActivitySim Dev Team,
I'm currently developing a model for Melbourne and want to implement ActivitySim. However, I do not clearly understand the description in the document. I have some questions as below:
1. Regarding the Skim matrix, can you help me explain what are these parameters and correct me if I'm wrong:
IVT - In-vehicle time, if this is train or bus so it is the time in bus, train
IWAIT - Initial wait time, the time the transporter waits at the station for the bus, train
XWAIT - Transfer wait time, the transfer time between trips
WACC - Walk access time, the time from their origin to the station
WAUX - Auxiliary walk time, the time the transporter travels from the current station to the next station
WEGR - Walk egress time, the travel time from the last station to the destination
TOTIVT - Total in-vehicle time, if the transporter uses a car, it is the sum of in-car time and in-train/bus time
KEYIVT - Transit submode in-vehicle time, I don't understand this, can you explain more?

2. In the example_estimation project, there are some modes such as EXP, LFR, and COM, and I don't want to use this but I can't find the disable/enable variable for these. Do I need to update the expression myself?

3. Is the example_estimation project enough for a general and simple model?

Regards,
Huy

**Comment by i-am-sijia** [comment]
Hi Huy,

Thanks for reaching out! 

1. KEYIVT is the IVT specific to the transit sub-modes. In the estimation example (the prototype_mtc model), the [transit sub-modes](https://github.com/ActivitySim/activitysim/blob/3d255d08c646c924c65e1453b401dda944f8802c/activitysim/examples/prototype_mtc/configs/tour_mode_choice.yaml#L42-L46) are: local bus (LOC), express bus (EXP), light rail / ferry (LRF), heavy rail (HVY), commuter rail (COM); they are then further separated by walk and drive access. The example model uses [KEYIVT > 0](https://github.com/ActivitySim/activitysim/blob/3d255d08c646c924c65e1453b401dda944f8802c/activitysim/examples/prototype_mtc/configs/tour_mode_choice_annotate_choosers_preprocessor.csv#L79) to check whether a specific transit sub-mode is available between the origin and destination, and then sets the [mode availability](https://github.com/ActivitySim/activitysim/blob/3d255d08c646c924c65e1453b401dda944f8802c/activitysim/examples/prototype_mtc/configs/trip_mode_choice.csv#L143) accordingly in the mode choice. IVT > 0 and TOTIVT >0 are not enough to tell us which transit sub-modes are available, because they are just the total IVT. The example model also applies a transit sub-mode specific IVT perception factor, such as [this one, e.g. every minute on commuter rail is perceived as only 0.7 mins by traveler](https://github.com/ActivitySim/activitysim/blob/3d255d08c646c924c65e1453b401dda944f8802c/activitysim/examples/prototype_mtc/configs/trip_mode_choice.csv#L145).

    If you are building a model with just simple, generic travel modes (i.e. car vs transit), then IVT and TOTIVT is enough for transit, but if you are building a model with sensitivities to transit sub-modes, then you want to know the IVT specific to transit sub-modes. Usually for regions with complex transit systems, transit sub-modes could be considered in the mode choice, which could be the case for Melbourne.

    KEYIVT is usually created during transit path building and skimming, modelers can ask the skimming application to save the IVT by transit modes, e.g. LOC_KEYIVT, EXP_KEYIVT, LRF_KEYIVT, ...

2. Do you want to remove transit sub-modes from mode choice? If so, you can delete their [corresponding COLUMNs, e.g., WALK_COM](https://github.com/ActivitySim/activitysim/blob/3d255d08c646c924c65e1453b401dda944f8802c/activitysim/examples/prototype_mtc/configs/trip_mode_choice.csv#L1) and corresponding expressions (ROWs) in the tour and trip mode choice UECs, and update the [nesting structure](https://github.com/ActivitySim/activitysim/blob/3d255d08c646c924c65e1453b401dda944f8802c/activitysim/examples/prototype_mtc/configs/tour_mode_choice.yaml#L42-L46) in the tour and trip mode choice configurations.

3. The prototype_mtc model is the general and simple model you can start with, and you can use example_estimation if you'd like to re-estimate any components.

Sijia

**Comment by HuyLeDinhGia** [comment]
Hi @i-am-sijia , 
Thank for your answer. 

Regards,
Huy
